core/src/main/java/org/apache/calcite/rel/rules/PruneEmptyRules.java

```diff
@@ -540,4 +540,32 @@ public interface JoinRightEmptyRuleConfig extends PruneEmptyRule.Config {
       };
     }
   }
+
+
+  public static final RelOptRule EMPTY_TABLE =
+      ImmutableEmptyTableOptimizationConfig.of()
+          .withOperandSupplier(b0 -> {
+            return b0.operand(TableScan.class).noInputs();
+          })
```

Comment:
I think it can be simplified to `.withOperandSupplier(b0 -> b0.operand(TableScan.class).noInputs());`